### DISCUSSION TITLE
Proper way to bypass assertPointerEvents in user.click

### DISCUSSION THREAD

**Discussion by make-github-pseudonymous-again** [testing-library/user-event#841] [Q&A] [answered]
While writing tests I often get into trouble trying to `click` elements that do not pass the [`hasPointerEvents`](https://github.com/testing-library/user-event/blob/a5ca2e47f30ea71b72a443c227ea125beb4e84b7/src/utils/pointer/cssPointerEvents.ts#L6-L21) check it embeds.

Most of the time, it means there is an issue with the UI code, not with `hasPointerEvents`. So far, I have encountered  only one situation where I could not fix the UI code and had to bypass this check. The only solution seems to be to define:

```typescript
const userWithoutPointerEventsCheck = userEvent.setup({
  pointerEventsCheck: PointerEventsCheckLevel.Never,
});
```

Would it make sense to allow passing the `pointerEventsCheck` option directly to `user.click` instead? Is there a deliberate choice to make it hard for users to bypass this check?

**Comment by ph-fritsche** [comment, marked as answer]
No it's not a deliberate choice to make it hard for users, but it was a deliberate choice to remove most options from the interaction APIs.

I don't think that this...
```js
const user = userEvent.setup()
await user.click(element) // click with pointer-events check
await user.setup({pointerEventsCheck: PointerEventsCheckLevel.Never}).click(element) // click without pointer-events check
await user.click(element) // continue with pointer-events check
```
... is harder to use than ...
```js
const user = userEvent.setup()
await user.click(element) // click with pointer-events check
await user.click(element, {pointerEventsCheck: PointerEventsCheckLevel.Never}) // click without pointer-events check
await user.click(element) // continue with pointer-events check
```

But the implementation code for the latter requires either an internal call to setup in every API or it requires passing down the options internally (as it was before) which results in a lot of copy-paste-code. It unnecessarily mixes concerns and results in repetitive code.
Such code tends to introduce bugs as both the code and the tests are more declarative than functional and reviewing those is more like proof-reading a config. Because that code would be all over the place, anybody working on any optional feature would have to remember to change it in every place.

**Reply by make-github-pseudonymous-again** [reply]
OK. Makes sense.

I did not know you could chain `setup` calls! Good to know.
